xds/src/main/java/io/grpc/xds/RouteMatch.java

```diff
@@ -60,7 +60,7 @@
    *
    * <p>Match is not deterministic if a runtime fraction match rule presents in this RouteMatch.
    */
-  boolean matches(String path, Map<String, Set<String>> headers) {
+  boolean matches(String path, Map<String, Iterable<String>> headers) {
```

Comment:
Why not just `boolean matches(String path, Metadata headers)` without making a copy map?  
```java
for (HeaderMatcher headerMatcher : headerMatchers) {
  headerName = headerMatcher.getName();
  Iterable<String> values = headers.getAll(
      Metadata.Key.of(headerName, Metadata.ASCII_STRING_MARSHALLER));
  if (headerMatcher.matchesValue(values)) {
    return false;
  }
}
```

Reply:
Seems the copy map can be reused for multiple RouteMatches in pickSubchannel(), which is more efficient.

Reply:
Yes, exactly.